packages/shared/pkg/proxy/proxy.go

```diff
@@ -33,10 +34,13 @@ func New(
 
 	return &Proxy{
 		Server: http.Server{
-			Addr:              fmt.Sprintf(":%d", port),
-			ReadTimeout:       0,
-			WriteTimeout:      0,
-			IdleTimeout:       idleTimeout,
+			Addr:         fmt.Sprintf(":%d", port),
+			ReadTimeout:  0,
+			WriteTimeout: 0,
+			// Downstream idle > upstream idle,
+			// otherwise a new downstream connection can try to reuse an open upstream connection
+			// and reuse TCP connection which is already closed
```

Comment:
```suggestion
			// otherwise a new downstream connection can try to reuse an upstream connection
			// which is already in state `CLOSE_WAIT` resulting in error
```